vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/payment/post_transaction_anon/post_transaction_anon.cc

```diff
@@ -0,0 +1,140 @@
+#include "bat/ledger/internal/request/request_util.h"
+#include "net/http/http_status_code.h"
+
+using std::placeholders::_1;
+
+namespace ledger {
+namespace endpoint {
+namespace payment {
+
+PostTransactionAnon::PostTransactionAnon(bat_ledger::LedgerImpl* ledger):
+    ledger_(ledger) {
+  DCHECK(ledger_);
+}
+
+PostTransactionAnon::~PostTransactionAnon() = default;
+
+std::string PostTransactionAnon::GetUrl(const std::string& order_id) {
+  const std::string path = base::StringPrintf(
+      "/v1/orders/%s/transactions/anonymousCard",
+      order_id.c_str());
+
+  return GetServerUrl(path);
+}
+
+std::string PostTransactionAnon::GeneratePayload(
+    const double amount,
+    const std::string& order_id,
+    const std::string& destination) {
+  base::Value denomination(base::Value::Type::DICTIONARY);
+  denomination.SetStringKey("amount", base::StringPrintf("%g", amount));
+  denomination.SetStringKey("currency", "BAT");
+
+  base::Value octets(base::Value::Type::DICTIONARY);
+  octets.SetKey("denomination", std::move(denomination));
+  octets.SetStringKey("destination", destination);
+
+  std::string octets_json;
+  base::JSONWriter::Write(octets, &octets_json);
+
+  const auto sign_headers = braveledger_request_util::GetSignHeaders(
+      order_id,
+      octets_json,
+      "primary",
+      ledger_->state()->GetRecoverySeed(),
+      true);
+
+  base::Value headers(base::Value::Type::DICTIONARY);
+  headers.SetStringKey("digest", sign_headers.at("digest"));
+  headers.SetStringKey("idempotency-key", order_id);
+  headers.SetStringKey("signature", sign_headers.at("signature"));
+
+  base::Value transaction(base::Value::Type::DICTIONARY);
+  transaction.SetKey("headers", std::move(headers));
+  transaction.SetStringKey("octets", octets_json);
+
+  std::string transaction_json;
+  base::JSONWriter::Write(transaction, &transaction_json);
+  std::string transaction_base64;
+  base::Base64Encode(transaction_json, &transaction_base64);
+
+  base::Value body(base::Value::Type::DICTIONARY);
+  body.SetStringKey("paymentId", ledger_->state()->GetPaymentId());
+  body.SetStringKey("kind", "anonymous-card");
+  body.SetStringKey("transaction", transaction_base64);
+
+  std::string json;
+  base::JSONWriter::Write(body, &json);
+  return json;
+}
+
+ledger::Result PostTransactionAnon::CheckStatusCode(const int status_code) {
+  if (status_code == net::HTTP_BAD_REQUEST) {
+    BLOG(0, "Invalid request");
+    return ledger::Result::LEDGER_ERROR;
+  }
+
+  if (status_code == net::HTTP_NOT_FOUND) {
+    BLOG(0, "Unrecognized transaction suffix");
+    return ledger::Result::NOT_FOUND;
+  }
+
+  if (status_code == net::HTTP_CONFLICT) {
+    BLOG(0, "External transaction id already submitted");
+    return ledger::Result::LEDGER_ERROR;
+  }
+
+  if (status_code == net::HTTP_INTERNAL_SERVER_ERROR) {
+    BLOG(0, "Internal server error");
+    return ledger::Result::LEDGER_ERROR;
+  }
+
+  if (status_code != net::HTTP_CREATED) {
```

Comment:
Just checking: can this endpoint ever return 200 for any reason? Do we want to treat 200 as an error?

Reply:
yes, only 201 is success